src/levelMap.js

```diff
@@ -55,13 +58,31 @@ function LevelMap(props) {
             });
     }
 
+    const fetchLevelMap = () => {
+        fetch(`${BASE_URL[environment]}/api/level?space=` + space, {
+            crossDomain: true,
+            method: 'GET',
+            headers: {
+                'Content-type': 'application/json'
+            }
+        })
+            .then(response => response.json())
+            .then(response => {
+                dispatch(setLevels(response))
+                setLevelMap(response.map(level => {
+                    return <div onClick={(e) => getLevel(e)}>{level}</div>
+                }))
+            })
+    }
+
+    fetchLevelMap()
```

Comment:
can we do this in one `useEffect` on mount? Otherwise it will get called everytime the compoenent re-renders, which should not happen too often, but just in case

Reply:
No it actually does happen infinitely, I used an explicit length check to do it now. Switching to `useEffect` will be better though?

Reply:
haan yes will def be better 
`useEffect(() => {
    fetchLevelMap()
}, []);` 
should do it